File: src/AgentIsland/Core/AppEnvironment.cs
namespace AgentIsland.Core;

public enum AppMode
{
    Normal,
    Demo,
    Debug,
}

/// Mode flags resolved once at launch. Demo injects synthetic data for
/// screenshots and must never fire real resume commands; the legacy
/// CODEXISLAND_* variables are still accepted as a fallback, matching macOS.
public static class AppEnvironment
{
    public static AppMode Current { get; } = Resolve();

    public static bool IsDemo => Current == AppMode.Demo;
    public static bool IsDebug => Current == AppMode.Debug;

    private static AppMode Resolve()
    {
        if (Flag("AGENTISLAND_DEMO") || Flag("CODEXISLAND_DEMO")) return AppMode.Demo;
        if (Flag("AGENTISLAND_DEBUG") || Flag("CODEXISLAND_DEBUG")) return AppMode.Debug;
        return AppMode.Normal;
    }

    private static bool Flag(string name)
    {
        var value = Environment.GetEnvironmentVariable(name);
        return value is "1" or "true" or "TRUE" or "yes";
    }
}

File: Sources/Model/AppEnvironment.swift
import Foundation

enum AppMode {
    case normal
    case demo
    case debug
}

enum AppEnvironment {
    static let current: AppMode = {
        let env = ProcessInfo.processInfo.environment
        func on(_ key: String) -> Bool { env[key] == "1" }
        if on("AGENTISLAND_DEMO") { return .demo }
        if on("AGENTISLAND_DEBUG") { return .debug }
        return .normal
    }()

    static var isDemo: Bool { current == .demo }
    static var isDebug: Bool { current == .debug }

    static var demoGuestFixturesEnabled: Bool {
        isDemo && ProcessInfo.processInfo.environment["AGENTISLAND_DEMO_GUESTS"] == "1"
    }
}
